plugins/autodoc/autodoc-plugin/src/main/java/org/eclipse/edc/plugins/autodoc/tasks/AutodocBomTask.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.plugins.autodoc.tasks;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.util.internal.GFileUtils;
+
+import java.io.File;
+
+public class AutodocBomTask extends DefaultTask {
+
+    public static final String NAME = "autodocBom";
+    public static final String DESCRIPTION = """
+            This task is intended for BOM modules. It resolves all autodoc manifests of modules that the BOM depends on
+            and generates a merged manifest file. By default, this merged file is stored at {project}/build/edc.json.
+            """;
+    private final JsonFileAppender appender;
+    private File outputFile;
+
+    public AutodocBomTask() {
+        appender = new JsonFileAppender(getLogger());
+        outputFile = getProject().getLayout().getBuildDirectory().file(outputFileName()).get().getAsFile();
+    }
+
+    @TaskAction
+    public void mergeManifests() {
+        if (!getProject().getName().endsWith("-bom")) {
+            getLogger().warn("Project name does not end with '-bom'. Is this really a BOM module?");
```

Comment:
maybe this functionality shouldn't be limited to bom modules, e.g. an EDC distribution (like Tractus-X) could be interested in generating a comprehensive autodoc out of a set of dependencies.
the warning wouldn't be needed then

Reply:
this would require a bit of refactoring, because then, the `autodocBom` task would not depend on the `resolveManifest` task anymore, but on either `resolveManifest` or `downloadManifests`.

alternatively, `resolveManifest` and `downloadManifest` could be merged into one, and artifact resolution (of the manifest file) happens dynamically based on its URI.

Definitely worth exploring, though in a different PR, IMO.